teammapper-backend/src/map/services/maps.service.ts

```diff
@@ -233,15 +239,13 @@ export class MapsService {
         'lastmodifiednode.nodeMapid = map.id'
       )
       .where(
-        // delete all maps that have nodes that were last updated after afterDays
         "(lastmodifiednode.lastUpdatedAt + (INTERVAL '1 day' * :afterDays)) < :today",
         { afterDays, today }
       )
       .orWhere(
         new Brackets((qb) => {
-          // also delete empty maps, use th emaps lastmodified date for this:
-          qb.where('lastmodifiednode.lastUpdatedAt IS NULL').andWhere(
-            "(map.lastModified + (INTERVAL '1 day' * :afterDays)) < :today",
+          qb.where("lastmodifiednode.lastUpdatedAt IS NULL").andWhere(
+            "(GREATEST(map.lastAccessed, map.lastModified) + (INTERVAL '1 day' * :afterDays)) < :today",
```

Comment:
is the "and" here correct? Usually, there shouldn't be any map without a node, therefore the first clause will be never null. So we might use it to collect broken maps but then it should be an or? Can you test this thought in a test case? I think most of the test do not include any nodes, so we are not really testing what is happening on real systems.

Reply:
I've added two new test cases now:

- It should delete a map where both lastAccessed and lastUpdatedAt are outdated
- It should not delete a map where lastAccessed is outdated but lastUpdatedAt on some nodes are recent

Both work just fine. But I agree in theory that it feels like it's at the wrong spot and the first clause should never be true.